# Capital Asset Pricing Model

The capital asset pricing model (CAPM) is a model used to determine a theoretically appropriate required rate of return of an asset.

The model takes into account the asset's sensitivity to non-diversifiable risk, as well as the expected rate of return of the market and the expected return of a theoretical risk-free asset. CAPM assumes a particular form of utility functions, where only first and second moments matter, that is, risk measured by variance, or asset returns whose probability distributions are completed described by the first two moments like the Gaussian distribution and zero transaction costs, which is necessary for diversification to remove all idiosyncratic risk.

The formula is given by:

$E(R_{i} = R_{f} + \beta_{i}(E(R_{m}) - R_{f}) \ni$

$E(R_{i})$ = capital asset expected return \
$R_{f}$ = risk-free rate of interest \
$\beta_{i}$ = sensitivity of expected excess asset returns to the expected excess market returns. \
$E(R_{m})$ = expected market return

Other derivations:

$\beta_{i}$ can also be expressed as $\dfrac{Cov(R_{i}, R_{m})}{Var(R_{m})}$, which is simply $p_{i, m}\dfrac{\sigma_{i}}{\sigma_{m}} \ni$

$E(R_{m}) - R_{f}$ = market premium \
$E(R_{i}) - R_{f}$ = risk premium \
$p_{i, m}$ = correlation coefficient between $i$ and $m$ \
$\sigma_{i}$ = standard deviation for i \
$\sigma_{m}$ = standard deviation for m

Restated in terms of risk premium, we find that individual risk premium is equal to the market premium times $\beta$:

$E(R_{i}) - R_{f} = \beta_{i}(E(R_{m}) - R_{f})$

Notes:
1. The expected market rate of return is usually estimated by measuring the arithmetic average of the historical returns on a market portfolio.
2. The risk-free ate of return used for determining the risk premium is usually the arithmetic average of the historical risk-free rates of return and not the current risk-free rate of return.

The meaning of $\beta$ is that when the expected market risk premium increases by 1\%, the individual asset's expected return increases by $\beta$\%, and vice versa. Since $\mu_{\beta} = 1$, whenever an asset's $\beta$ is greater than 1. the market risk is higher than that of the average asset.

In [2]:
import numpy as np
import statsmodels.api as sm

In [3]:
y = [1, 2, 3, 4, 2, 3, 4]
x = range(1, 8)

x = sm.add_constant(x)

results = sm.OLS(y, x).fit()
print(results.params)
results.summary()

[1.28571429 0.35714286]


/usr/local/anaconda3/lib/python3.7/site-packages/statsmodels/stats/stattools.py:71: ValueWarning: omni_normtest is not valid with less than 8 observations; 7 samples were given.
  "samples were given." % int(n), ValueWarning)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.481
Model:                            OLS   Adj. R-squared:                  0.377
Method:                 Least Squares   F-statistic:                     4.630
Date:                Sat, 31 Oct 2020   Prob (F-statistic):             0.0841
Time:                        21:22:29   Log-Likelihood:                -7.8466
No. Observations:                   7   AIC:                             19.69
Df Residuals:                       5   BIC:                             19.59
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          1.2857      0.742      1.732      0.144      -0.622       3.194
x1             0.3571      0.166      2.152      0.084      -0.070       0.784
==============================================================================
Omnibus:                          nan   Durbin-Watson:                   1.976
Prob(Omnibus):                    nan   Jarque-Bera (JB):                0.342
Skew:                           0.289   Prob(JB):                        0.843
Kurtosis:                       2.083   Cond. No.                         10.4
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

/usr/local/anaconda3/lib/python3.7/site-packages/statsmodels/stats/stattools.py:71: ValueWarning: omni_normtest is not valid with less than 8 observations; 7 samples were given.
  "samples were given." % int(n), ValueWarning)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.481
Model:                            OLS   Adj. R-squared:                  0.377
Method:                 Least Squares   F-statistic:                     4.630
Date:                Fri, 30 Oct 2020   Prob (F-statistic):             0.0841
Time:                        21:34:06   Log-Likelihood:                -7.8466
No. Observations:                   7   AIC:                             19.69
Df Residuals:                       5   BIC:                             19.59
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          1.2857      0.742      1.732      0.144      -0.622       3.194
x1             0.3571      0.166      2.152      0.084      -0.070       0.784
==============================================================================
Omnibus:                          nan   Durbin-Watson:                   1.976
Prob(Omnibus):                    nan   Jarque-Bera (JB):                0.342
Skew:                           0.289   Prob(JB):                        0.843
Kurtosis:                       2.083   Cond. No.                         10.4
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Since the $\beta$ has a $t$ of 2.152, which is greater than 2, we could claim that it is significantly 